core/src/main/java/fr/sncf/osrd/envelope_sim/allowances/AbstractAllowanceWithRanges.java

```diff
@@ -309,22 +307,55 @@ public Envelope computeIteration(Envelope base,
 
         // 1) compute the potential junction parts (slowdown or speedup)
         var leftPart = computeLeftJunction(base, coreEnvelope, imposedBeginSpeed);
+
         var leftPartEndPos = leftPart != null ? leftPart.getEndPos() : base.getBeginPos();
         var coreEnvelopeWithLeft = computeEnvelopeWithLeftJunction(base, coreEnvelope, leftPart);
 
         var rightPart = computeRightJunction(base, coreEnvelopeWithLeft, imposedEndSpeed);
         var rightPartBeginPos = rightPart != null ? rightPart.getBeginPos() : base.getEndPos();
 
-        // if the junction parts touch or intersect, there is no core phase
-        if (rightPartBeginPos <= leftPartEndPos) {
+        if (EnvelopePhysics.areIntersecting(rightPart, leftPart)) {
+            // if the junction parts touch or intersect, there is no core phase
             return intersectLeftRightParts(leftPart, rightPart);
```

Comment:
I think it's pretty janky to attempt to detect intersection first using one method, then use another method to perform it. it's bound to create bugs, and we should either not merge this, or file a bug to fix this, and not forget about it.